### DISCUSSION TITLE
id dont match but still update facing issue in realtime subscribe

### DISCUSSION THREAD

**Discussion by apujayed** [pocketbase/pocketbase#3562] [Q&A] [answered]
```
 useEffect(() => {
        if (current_catalog !== undefined) {
            pb.collection('auction_info').subscribe('*', function (e) {
                console.log(e.record.catalog);
                if (e.record.catalog === current_catalog.id) {  
                    set_highest(e.record.price_max)
                    set_current_bidder(e.record.bidder_current)
                }
            });
            set_pkgs(current_catalog.Package)
            set_catalog_id(current_catalog.id)
            set_offer_price(current_catalog.Offer_price)
            set_lot_number(current_catalog.Lot_number)
            set_invoice_number(current_catalog.Invoice)
            set_grade(current_catalog.Grade);
            (async () => {
                try {
                    set_Offer_price(current_catalog.Offer_price)
                    const FactoryRecord = await pb.collection('profiles').getFirstListItem(`id="${current_catalog.Factory}"`, {
                        expand: 'relField1,relField2.subRelField',
                    });
                    set_factory(FactoryRecord.Company_name)
                }
                catch (error) {
                    // 
                }
                // console.log()
             
            })()
        }
    }, [current_catalog])
```
here current catalog change every one second

**Comment by ganigeorgiev** [comment, marked as answer]
I'm not sure that I understand what exactly is the intended behavior of the above code.

At first glance it seems that it will create a new subscription every time when `current_catalog` changes. If you really need to do this, you should consider unsubscribing from the previous subscriptions. The `subscribe()` call returns a Promise with the unsubscribe function that you can call on the effect destroy. Alternative you can also call `pb.collection("auction_info").unsubscribe("*")` to unsubscribe from all auction_info `*` subscriptions.

It will depend on your actual code, where it runs and how, but another option could be to register the subscription **outside** of the effect because at the end the callback of the subscribe call will be triggered whenever the related PocketBase server data has changed (aka. `current_catalog` will be most likely already changed client-side).

_Side-note: Filtering by id like `pb.collection('profiles').getFirstListItem('id=...', ...)` doesn't really make sense unless you want to target specifically the List API rule of the collection. Usually a more efficient request when searching directly by id is to call `getOne(current_catalog.Factory, ...)` instead._
